packages/@aws-cdk/core/lib/private/synthesis.ts

```diff
@@ -103,10 +105,20 @@ function prepareTree(root: IConstruct) {
  * Stop at Assembly boundaries.
  */
 function synthesizeTree(root: IConstruct, builder: cxapi.CloudAssemblyBuilder) {
-  visit(root, 'post', construct => construct.onSynthesize({
-    outdir: builder.outdir,
-    assembly: builder,
-  }));
+  visit(root, 'post', construct => {
+    const session = {
+      outdir: builder.outdir,
+      assembly: builder,
+    };
+
+    if (construct instanceof Stack) {
+      construct._synthesizeTemplate(session);
```

Comment:
So, theoretically, if a user implemented `synthesize` on the stack level, we won't be calling it anymore - right? That sounds like a breaking change, am I missing something?

Reply:
We are still calling `synthesize` through the `onSynthesize()` shim.

Reply:
Not sure I follow - where? I see we are calling `onSynthesize` on constructs that aren't a `stack`, but what about those who are?

Reply:
Mmm. okay let me double check.

Reply:
Okay, nice catch!